app/main/db/driver.js

```diff
@@ -7,6 +7,16 @@ const driver = {
   getConnection() {
     return connection
   },
+  async getStats() {
```

Comment:
@oferitz @ ronzeidman
Perhaps you can help me out - what am I doing wrong?

Reply:
both `r.` and `connection.` don't really work

Reply:
you have to run the query `await r.table('stam').get('80b4ce56-659e-48a2-a518-75be1b53a5d0').run(connection)`

Reply:
Awesome it works, thanks @ronzeidman. Isn't that a bit cumbersome though. Do you think the user should have his own way of negating the need to repeat this?

Reply:
if you use a connection pool you can drop the `connection` part of it. but the problem of awaiting a query directly is that it's misleading users to think that chaining a query runs it since they may think it's already a promise and a promise eagerly evaluates.

Reply:
therefore people may think that saving a query and running it several times shouldn't work, or saving a part of a query and chaining different ends wouldn't work, and that's not the case.

Reply:
in RethinkDBdash it works but it's because the query just implements `.then` function it's not a real promise. treating a query sometimes as a promise and sometimes as a query is misleading since a query is not a promise and therefore you can:
`.run()` as many times as you want (promises run only once and return the same value without running other times)
be stored for future evaluation (promises run as you create them)

Reply:
a promise would work like this:
```
const func1Promise = func1();
await func2();
await thePromise;
await thePromise;
```
in this scenerio func1 would start evaluating before func2 and run only once.
now look at this:
```
const query = r.table(...);
await func2();
await query;
await query;
```
the query will evaluate twice and only after func2 has finished evaluating

Reply:
Interesting, I see what you mean now. Come to think of it adding `.filter` to something that looks like a promise is weird. It is indeed misleading, though I never really had any issues.

Reply:
"Its never an issue until it's an issue" these issues can be subtle and hard to debug, that why I like to avoid them when possible

Reply:
I Wish this driver had a options pool :(

Reply:
@thelinuxlich it does. Just `await r.connectPool(options)`
You can see the deviations from rethinkdbdash in my readme file. It's actually almost the same

Reply:
great Ron, your driver will be the reference for the other languages!